modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/factory/CustomElementCETImplFactoryImpl.java

```diff
@@ -113,31 +123,43 @@ public void validate(
 			}
 			else {
 				throw new ClientExtensionEntryTypeSettingsException(
-					"html-element-name-contains-invalid-character-x", c);
+					"Custom element HTML element name contains an" +
+					"invalid character: " + c,
+					"html-element-name-contains-invalid-character-x", c
+				);
 			}
 		}
 
 		if (!containsDash) {
 			throw new ClientExtensionEntryTypeSettingsException(
-				"html-element-name-must-contain-at-least-one-hyphen");
+				"Custom element HTML element name must contain at least one " +
+				"hyphen",
+				"html-element-name-must-contain-at-least-one-hyphen"
+			);
 		}
 
 		if (_reservedHTMLElementNames.contains(htmlElementName)) {
 			throw new ClientExtensionEntryTypeSettingsException(
-				"x-is-a-reserved-html-element-name", htmlElementName);
+				"Reserved custom element HTML element name " + htmlElementName,
+				"x-is-a-reserved-html-element-name", htmlElementName
+			);
 		}
 
 		String urls = newCustomElementCET.getURLs();
 
 		if (Validator.isNull(urls)) {
 			throw new ClientExtensionEntryTypeSettingsException(
-				"please-enter-at-least-one-url");
+				"Please enter at least one URL",
+				"please-enter-at-least-one-url"
+			);
 		}
 
 		for (String url : urls.split(StringPool.NEW_LINE)) {
 			if (!Validator.isUrl(url, true)) {
 				throw new ClientExtensionEntryTypeSettingsException(
-					"url-x-is-invalid", url);
+					"Invalid custom element URL " + url,
```

Comment:
Missing a colon to match other cases and adding "custom element" is probably not necessary: `Invalid URL: `